### PR TITLE
Allow workbench to load the workspaces from mantidplot projects

### LINKED ISSUE #27766: Load MantidPlot projects into Workbench
Some users have not switched to workbench as they are unable to load their work from MantidPlot. 

We should have some sort of conversion, so MantidPlot projects can be loaded into workbench.

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**

Allows workbench to load the workspaces from mplot projects.
If the file fails to load as JSON, then it will fall back to a second option of selectively parsing out the workspace liost from a mantidplot project.

It does not attempt to do graphs or anything else from the project.  Also qitplot tables will not be recovered.

**Report to:** @AnthonyLim23 

**To test:**
1. Create a project in mantidplot and save it, not the workspaces it includes.
1. Try to load it in workbench, the workspaces should load.

Fixes #27766 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v4.3.0/mantidworkbench.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ MantidWorkbench Changes
 New
 ###
 - Waterfall Plots
+- Mantid Workbench can now load all of the workspaces from a projects saved from Mantidplot.  Graphs and interface values are not imported from the project.
```
Comment (ConorMFinn) [inline docs/source/release/v4.3.0/mantidworkbench.rst]: Typo
```suggestion
- Mantid Workbench can now load all of the workspaces from projects saved from Mantidplot.  Graphs and interface values are not imported from the project.
```

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just a typo in the docs that needs correcting. I also noticed that the loading does not maintain group workspaces, but I assume that is the expected behavior.

**Review by cailafinn — APPROVED** [review]
This looks good to me now. Code looks good and all the saved workspaces from MantidPlot maintain their workspace type and name.

**Comment by gemmaguest** [comment]
It was requested to report this to @AnthonyLim23 when merged

**Pull request merged**